frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/ToggleListCommitsAction.java

```diff
@@ -14,12 +15,26 @@
 /**
  * Expects DataKeys:
  * <ul>
+ *  <li>{@link DataKeys#KEY_BRANCH_LAYOUT}</li>
  *  <li>{@link DataKeys#KEY_GRAPH_TABLE_MANAGER}</li>
  * </ul>
  */
 public class ToggleListCommitsAction extends ToggleAction implements DumbAware {
   private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("frontendActions");
 
+  @Override
+  @UIEffect
+  public void update(AnActionEvent e) {
+    super.update(e);
+
+    var branchLayout = getBranchLayout(e);
+    if (branchLayout.isDefined()) {
+      boolean anyChildBranchExists = branchLayout.get().getRootEntries()
+          .exists(rootBranch -> rootBranch.getSubentries().nonEmpty());
+      e.getPresentation().setEnabled(anyChildBranchExists);
```

Comment:
Hmmm :thinking: what about setting some description that will explain to the user why the action is disabled?

Reply:
Done